### DISCUSSION TITLE
Application Insights Instrumentation Key

### DISCUSSION THREAD

**Discussion by DaleyKD** [pnp/pnpcore#488] [Q&A] [answered]
What's the purpose of the hardcoded Instrumentation Key in `TelemetryManager`?  Does that mean all telemetry is going to Microsoft and not to my Application Insights?

**Comment by jansenbe** [comment, marked as answer]
It's indeed hardcoded as it optionally sends information about which models are loaded, the idea to use this data to better prioritize future Microsoft Graph investments. If you don't want that behavior, just turn off telemetry.

**Reply by DaleyKD** [reply]
Exactly what was needed to know.  It's to send data to Microsoft.  Thanks!
